src/web/module/activity/component/View.tsx

```diff
@@ -0,0 +1,107 @@
+import { DateTime } from 'luxon';
+import { useState, useEffect } from 'react';
+
+import { History } from '@/api/module/history/entity';
+import { HistoryStat } from '@/api/module/stat/entity';
+import { getHistoryStat } from '@/api/module/stat/service';
+import { getTotalJuzFromLines } from '@/api/shared/entity/juz';
+import { useData } from '@/web/module/activity/context/DataContext';
+
+interface InternalProps {
+  mapHistoryStats: Map<string, HistoryStat>;
+  currentDate: Date;
+}
+
+const CLASS_NAMES: Record<string, string> = {
+  container: 'flex justify-between',
+  content: 'text-lg text-custom-teal',
+};
+
+const formatDate = (date: Date): string => {
+  const parsedTime: DateTime = DateTime.fromJSDate(date);
+  if (!parsedTime.isValid) return '';
+  return parsedTime.toFormat('yyyy-MM-dd EEE');
+};
+
+const updateAndRenderCurrentDate = (i: InternalProps, item: History): JSX.Element => {
+  if (i.currentDate) {
+    const parsedCurrentDate: DateTime = DateTime.fromJSDate(i.currentDate);
+    const parsedOccuredAt: DateTime = DateTime.fromJSDate(item.occuredAt);
+    if (parsedCurrentDate.hasSame(parsedOccuredAt, 'day')) return <></>;
+  }
+
+  i.currentDate = item.occuredAt;
```

Comment:
Any other concern for this? 

Because we will use this design a lot since we took out the function out from exported component. And this wouldn't need param back then when it still inside exported component, now we use this InternalProps approach and pass it to any function that needed as inside function replacement.

For the changes of value itself this is the way that I know so far the achieved the result. Later on will move this logic to api instead.

But if in somecase we need `let` variable and pass also change the value in other function, any better way to implement it?